Automattic/pocket-casts-ios
# Playback 2023: add animations to some stories

| 📘 Part of: #1142 | 🎨 Designs: [Figma](https://www.figma.com/file/lH66LwxxgG8btQ8NrM0ldx/End-of-Year?type=design&node-id=1599-20385&mode=design) |
|:---:|:---:|

Animates the following stories:

*Listened time (2nd story)*

https://github.com/Automattic/pocket-casts-ios/assets/7040243/1e5b37c4-f7e2-4b4a-85cb-4868682ce80a

*Top one podcast to top five podcast (third to fourth story)*

https://github.com/Automattic/pocket-casts-ios/assets/7040243/28f175de-46d2-4678-a1ed-5a197d76b252

*Longest episode (seventh story)*

https://github.com/Automattic/pocket-casts-ios/assets/7040243/357acefb-440c-47a5-b76c-11ee5535bc91

## To test

### Testing animations

1. Make sure you're logged in to an account that has a few episodes listened
2. Go to Profile > Settings > Beta Features > enable `endOfYear`
3. Go back to Profile and tap the End of Year card
4. Go to the second story
5. ✅ The two rows of podcasts should move
6. Tap the story and hold (to pause it)
7. ✅ The animation should pause
8. ✅ Untap and the animation should resume
9. Skip to the next story and then skip again
10. ✅ A transition from the top 1 to top 5 should happen (this is not pausable)
11. Go to the longest episode story
12. ✅ The covers should slide into the screen
13. ✅ Pause/resume the story and the animation should follow

### Testing sharing

1. For each story that is animated, tap "Share this story" and then "Save Image"
2. Check the generated asset in Photos
3. ✅ The story should appear correctly

Please test on a big device (iPhone 15 Pro Max, for example), a normal device (iPhone 15), and a smaller device (iPod touch).

## Checklist
